snuba/admin/views.py

```diff
@@ -17,6 +18,31 @@ def health() -> Response:
     return Response("OK", 200)
 
 
+@application.route("/clickhouse_queries")
+def clickhouse_queries() -> Response:
+    res = [q.to_json() for q in SystemQuery.all_queries()]
+    return make_response(jsonify(res), 200)
+
+
+@application.route("/run_clickhouse_system_query", methods=["POST"])
+def clickhouse_system_query() -> Response:
+    # TODO: You can do something like this to get all the hosts:
+    # SELECT * FROM system.clusters
+    req = request.get_json()
+    results, columns = run_system_query(
+        req.get("host", "localhost"),
```

Comment:
I think we should avoid letting the client pick any host they like and running the query there. Ideally the host and port values get picked up from the cluster that is configured in settings.

Reply:
But we want to be able to do system queries on different nodes right?

Reply:
I don't think we should let the client pick the host either.
Why cannot the client pick the storage or storage set and we resolve the cluster from there.
Of course we would have to fix our config to have all clusters properly defined. Which is something we should do anyway.
Then in order to identify the host you can pass shard and replica number. Or even the name at that point, so we can validate.